Failed tests with nbconvert-5.3.1 and pandoc-2.2.1

When building the package on openSUSE, I get these failed tests.
```
[   97s] =================================== FAILURES ===================================
[   97s] ________________________ TestMarkdown.test_markdown2rst ________________________
[   97s] 
[   97s] self = <nbconvert.filters.tests.test_markdown.TestMarkdown testMethod=test_markdown2rst>
[   97s] 
[   97s]     @dec.onlyif_cmds_exist('pandoc')
[   97s]     def test_markdown2rst(self):
[   97s]         """markdown2rst test"""
[   97s]     
[   97s]         #Modify token array for rst, escape asterik
[   97s]         tokens = copy(self.tokens)
[   97s]         tokens[0] = r'\*test'
[   97s]         tokens[1] = r'\*\*test'
[   97s]     
[   97s]         for index, test in enumerate(self.tests):
[   97s]             self._try_markdown(
[   97s]                 partial(
[   97s]                     convert_pandoc, from_format='markdown', to_format='rst'),
[   97s] >               test, tokens[index])
[   97s] 
[   97s] /usr/lib/python2.7/site-packages/nbconvert/filters/tests/test_markdown.py:228: 
[   97s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   97s] /usr/lib/python2.7/site-packages/nbconvert/filters/tests/test_markdown.py:233: in _try_markdown
[   97s]     self.assertIn(tokens, results)
[   97s] E   AssertionError: '\\*\\*test' not found in u'\\**test'
[   97s] _____________________ TestMarkdown.test_pandoc_extra_args ______________________
[   97s] 
[   97s] self = <nbconvert.filters.tests.test_markdown.TestMarkdown testMethod=test_pandoc_extra_args>
[   97s] 
[   97s]     @dec.onlyif_cmds_exist('pandoc')
[   97s]     def test_pandoc_extra_args(self):
[   97s]         # pass --no-wrap
[   97s]         s = '\n'.join([
[   97s]             "#latex {{long_line | md2l(['--no-wrap'])}}",
[   97s]             "#rst {{long_line | md2r(['--columns', '5'])}}",
[   97s]         ])
[   97s]         long_line = ' '.join(['long'] * 30)
[   97s]         env = Environment()
[   97s]         env.filters.update({
[   97s]             'md2l': lambda code, extra_args: convert_pandoc(
[   97s]                 code, from_format='markdown', to_format='latex',
[   97s]                 extra_args=extra_args),
[   97s]             'md2r': lambda code, extra_args: convert_pandoc(
[   97s]                 code, from_format='markdown', to_format='rst',
[   97s]                 extra_args=extra_args),
[   97s]         })
[   97s]         tpl = env.from_string(s)
[   97s]         rendered = tpl.render(long_line=long_line)
[   97s]         _, latex, rst = rendered.split('#')
[   97s]     
[   97s] >       self.assertEqual(latex.strip(), 'latex %s' % long_line)
[   97s] E       AssertionError: u'latex' != 'latex long long long long long long long long long long long long long long long long long long long long long long long long long long long long long long'
[   97s] 
[   97s] /usr/lib/python2.7/site-packages/nbconvert/filters/tests/test_markdown.py:103: AssertionError
[   97s] ----------------------------- Captured stderr call -----------------------------
[   97s] --no-wrap has been removed.  Use --wrap=none instead.
[   97s] Try pandoc --help for more information.
[   97s] =============================== warnings summary ===============================
[   97s] nbconvert/exporters/tests/test_notebook.py::TestNotebookExporter::test_downgrade_2
[   97s]   /usr/lib/python2.7/site-packages/nbformat/validator.py:251: UserWarning: No schema for validating v2 notebooks
[   97s]     warnings.warn("No schema for validating v%s notebooks" % version, UserWarning)
[   97s] 
[   97s] nbconvert/filters/tests/test_datatypefilter.py::TestDataTypeFilter::test_junk_types
[   97s]   /usr/lib/python2.7/site-packages/nbconvert/filters/datatypefilter.py:41: UserWarning: Your element with mimetype(s) ['hair', 'water', 'rock'] is not able to be represented.
[   97s]     mimetypes=output.keys())
[   97s] 
[   97s] nbconvert/filters/tests/test_datatypefilter.py::TestDataTypeFilter::test_null
[   97s]   /usr/lib/python2.7/site-packages/nbconvert/filters/datatypefilter.py:41: UserWarning: Your element with mimetype(s) [] is not able to be represented.
[   97s]     mimetypes=output.keys())
[   97s] 
[   97s] -- Docs: http://doc.pytest.org/en/latest/warnings.html
[   97s] == 2 failed, 225 passed, 7 skipped, 1 deselected, 3 warnings in 77.36 seconds ==
[   97s] error: Bad exit status from /var/tmp/rpm-tmp.78ZN5Y (%check)
[   97s] 
[   97s] 
[   97s] RPM build errors:
[   97s]     Bad exit status from /var/tmp/rpm-tmp.78ZN5Y (%check)
[   97s] 
[   97s] milic failed "build python-jupyter_nbconvert-doc.spec" at Thu Jul 26 08:32:48 UTC 2018.
[   97s] 

The buildroot was: /var/tmp/build-root/openSUSE_Tumbleweed-x86_64
matej@milic: python-jupyter_nbconvert$
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -24,10 +24,8 @@ addons:
     - cm-super # more fonts
     - texlive-xetex # latex to pdf converter
     - inkscape # for svgs in pdf output
-before_install:
-    - git clone --quiet --depth 1 https://github.com/minrk/travis-wheels travis-wheels
 install:
-    - wget https://github.com/jgm/pandoc/releases/download/1.19.1/pandoc-1.19.1-1-amd64.deb && sudo dpkg -i pandoc-1.19.1-1-amd64.deb
+    - wget https://github.com/jgm/pandoc/releases/download/2.7/pandoc-2.7-1-amd64.deb && sudo dpkg -i pandoc-2.7-1-amd64.deb
     - pip install --upgrade setuptools pip pytest
     - pip install -f travis-wheels/wheelhouse . codecov coverage
     - pip install nbconvert[execute,serve,test]

--- a/nbconvert/filters/tests/test_markdown.py
+++ b/nbconvert/filters/tests/test_markdown.py
@@ -231,10 +231,10 @@ i.e. the $i^{th}$"""
     def test_markdown2rst(self):
         """markdown2rst test"""
 
-        #Modify token array for rst, escape asterik
+        #Modify token array for rst, escape asterisk
         tokens = copy(self.tokens)
         tokens[0] = r'\*test'
-        tokens[1] = r'\*\*test'
+        tokens[1] = r'\**test'
 
         for index, test in enumerate(self.tests):
             self._try_markdown(

--- a/nbconvert/utils/pandoc.py
+++ b/nbconvert/utils/pandoc.py
@@ -15,7 +15,7 @@ from ipython_genutils.py3compat import cast_bytes, which
 from .exceptions import ConversionException
 
 _minimal_version = "1.12.1"
-_maximal_version = "2.0.0"
+_maximal_version = "3.0.0"
 
 def pandoc(source, fmt, to, extra_args=None, encoding='utf-8'):
     """Convert an input string using pandoc.

--- a/nbconvert/utils/pandoc.py
+++ b/nbconvert/utils/pandoc.py
@@ -17,6 +17,7 @@ from .exceptions import ConversionException
 _minimal_version = "1.12.1"
 _maximal_version = "3.0.0"
 
+
 def pandoc(source, fmt, to, extra_args=None, encoding='utf-8'):
     """Convert an input string using pandoc.
 
@@ -94,6 +95,9 @@ def check_pandoc_version():
     PandocMissing
       If pandoc is unavailable.
     """
+    if check_pandoc_version._cached is not None:
+        return check_pandoc_version._cached
+
     v = get_pandoc_version()
     if v is None:
         warnings.warn("Sorry, we cannot determine the version of pandoc.\n"
@@ -102,6 +106,7 @@ def check_pandoc_version():
                       RuntimeWarning, stacklevel=2)
         return False
     ok = check_version(v, _minimal_version, max_v=_maximal_version)
+    check_pandoc_version._cached = ok
     if not ok:
         warnings.warn( "You are using an unsupported version of pandoc (%s).\n" % v +
                        "Your version must be at least (%s) " % _minimal_version +
@@ -109,6 +114,9 @@ def check_pandoc_version():
                        "Refer to http://pandoc.org/installing.html.\nContinuing with doubts...",
                        RuntimeWarning, stacklevel=2)
     return ok
+
+
+check_pandoc_version._cached = None
 
 #-----------------------------------------------------------------------------
 # Exception handling

--- a/nbconvert/utils/tests/test_pandoc.py
+++ b/nbconvert/utils/tests/test_pandoc.py
@@ -28,6 +28,10 @@ class TestPandoc(TestsBase):
         super(TestPandoc, self).__init__(*args, **kwargs)
         self.original_env = os.environ.copy()
 
+    def setUp(self):
+        super(TestPandoc, self).setUp()
+        pandoc.check_pandoc_version._cached = None
+
     @onlyif_cmds_exist('pandoc')
     def test_pandoc_available(self):
         """ Test behaviour that pandoc functions raise PandocMissing as documented """
@@ -55,11 +59,16 @@ class TestPandoc(TestsBase):
 
         pandoc._minimal_version = "120.0"
         with warnings.catch_warnings(record=True) as w:
+            # call it twice to verify the cached value is used
             assert not pandoc.check_pandoc_version()
+            assert not pandoc.check_pandoc_version()
+        # only one warning after two calls, due to cache
         self.assertEqual(len(w), 1)
-
+        # clear cache
+        pandoc.check_pandoc_version._cached = None
         pandoc._minimal_version = pandoc.get_pandoc_version()
         assert pandoc.check_pandoc_version()
+
 
 def pandoc_function_raised_missing(f, *args, **kwargs):
     try:

